database/buildSchema/45_activity_log.sql

```diff
@@ -0,0 +1,383 @@
+                                WHERE
+                                    id = NEW.assigner_id)), 'stage', json_build_object('number', NEW.stage_number, 'name', (
+                                    SELECT
+                                        title FROM template_stage
+                                    WHERE
+                                        id = NEW.stage_id)), 'sections', (
+                                    SELECT
+                                        json_agg(t)
+                                        FROM (
+                                            SELECT
+                                                title, code, INDEX FROM template_section
+                                            WHERE
+                                                id = ANY (ARRAY ( SELECT DISTINCT
+                                                    template_section_id FROM review_question_assignment_section
+                                                WHERE
+                                                    review_assignment_id = NEW.id)) ORDER BY INDEX) t), 'review_level', NEW.level_number));
+    RETURN NULL;
+END;
+$application_event$
+LANGUAGE plpgsql;
+
+-- For this one we need to wait until the Trigger/Action is finished so we can check the newly-created review_question_assignment records
+CREATE TRIGGER assignment_activity_trigger
+    AFTER UPDATE ON public.review_assignment
+    FOR EACH ROW
+    WHEN (NEW.trigger IS NULL AND OLD.trigger = 'PROCESSING')
+    EXECUTE FUNCTION assignment_activity_log ();
+
+-- REVIEW STATUS CHANGES
+CREATE OR REPLACE FUNCTION public.review_status_activity_log ()
+    RETURNS TRIGGER
+    AS $application_event$
+DECLARE
+    app_id integer;
+    reviewer_id integer;
+    rev_assignment_id integer;
+    stage_number integer;
+    prev_status varchar;
+    level_num integer;
+    is_last_level boolean;
+    is_final_decision boolean;
+BEGIN
+    SELECT
+        r.application_id,
+        r.reviewer_id,
+        r.review_assignment_id,
+        r.stage_number,
+        r.level_number,
+        r.is_last_level,
+        r.is_final_decision INTO app_id,
+        reviewer_id,
+        rev_assignment_id,
+        stage_number,
+        level_num,
+        is_last_level,
+        is_final_decision
+    FROM
+        review r
+    WHERE
+        id = NEW.review_id;
+    prev_status = (
+        SELECT
+            status
+        FROM
+            review_status_history
+        WHERE
+            review_id = NEW.review_id
+            AND is_current = TRUE);
+    INSERT INTO public.activity_log (type, value, application_id, "table", record_id, details)
+        VALUES ('REVIEW', (
+                CASE WHEN NEW.status = 'DRAFT'
+                    AND prev_status IS NULL THEN
+                    'Started'
+                WHEN NEW.status = 'DRAFT'
+                    AND prev_status = 'CHANGES_REQUESTED' THEN
+                    'Re-started'
```

Comment:
Would this one also captures when the review was submitted? are you waiting on the decision of what to do about the review_decisions?

Reply:
It just passed the enum "SUBMITTED" straight through to value when submitting. I've added front-end logic to distinguish between initial submit and re-submit, it's a bit messy to do it here ([hence my question ](https://github.com/openmsupply/application-manager-web-app/issues/1090)about the "RESUBMITTED" status not being used -- this would be easier if that was being used)